view_user_birthdate.py

```diff
@@ -0,0 +1,251 @@
+"""Build a dictionary of usernames and User objects. Additionally, take
+in user input, match it to any usernames in the dictionary and return
+that User's formatted birthdate based on the User's nationality.
+"""
+import re
+from get_input import get_input
+
+REGEX_DATE_RULE = ("(?P<month>[0-9]{1,2})" # Month.
+                   "/(?P<day>[0-9]{1,2})" # Day.
+                   "/(?P<year>[0-9]{4})" # Year.
+                  )
+
+class Date(object):
+    """Take in a string in the format:
+    mm/dd/yyyy
+    Parse it and create a Date object with the string attributes
+    'month', 'day', and 'year'. A Date object can return its attributes
+    using its display_date() method.
+    """
+    def __init__(self, string):
+        """Instantiate a Date object with the attributes 'month',
+        'day', and 'year'.
+        """
+        assert_response = "A string in the format mm/dd/yyyy must be provided."
+        assert re.search(REGEX_DATE_RULE, string), assert_response
+        self._month, self._day, self._year = parse_date(string)
+
+    def get_date_attributes(self):
```

Comment:
In the above, you can see I made a private method - this is fine for *parents* to call their *children's* private methods. The reverse is not true, and in either case of private variables, that also violates encapsulation.

The parent, through the 'abstractmethod' decorator, is demanding that anyone who implements it *must* provide that function. It's a kind of contract, though not the same as assertions. Since it's demanding all children implement that function, the parent is safe to call that function and does not violate encapsulation by doing so.